libs/SmartSync/SmartSync/Classes/Manager/SFSmartSyncSyncManager.m

```diff
@@ -209,6 +223,11 @@ - (SFSyncState*) syncDownWithTarget:(SFSyncDownTarget*)target options:(SFSyncOpt
 /** Resync
  */
 - (SFSyncState*) reSync:(NSNumber*)syncId updateBlock:(SFSyncSyncManagerUpdateBlock)updateBlock {
+    if (syncId.integerValue == self.syncIdRunning) {
```

Comment:
Seems like this would be a good base for implementing a reasonable `isEqual:` on `SFSyncState`, and instead of having a single `syncIdRunning` property for the sync manager, have a map of `SFSyncState` objects to their running sync IDs.  Then you could have a similar comparison, and keep the ability to run multiple syncs at a time.